salt/modules/nilrt_ip.py

```diff
@@ -522,17 +549,7 @@ def _change_state(interface, new_state):
     :rtype: bool
     '''
     if __grains__['lsb_distrib_id'] == 'nilrt':
-        initial_mode = _get_adapter_mode_info(interface)
-        _save_config(interface, 'Mode', 'TCPIP' if new_state == 'up' else 'Disabled')
-        if initial_mode == 'ethercat':
-            __salt__['system.set_reboot_required_witnessed']()
-        else:
-            out = __salt__['cmd.run_all']('ip link set {0} {1}'.format(interface, new_state))
-            if out['retcode'] != 0:
-                msg = 'Couldn\'t {0} interface {1}. Error: {2}'.format('enable' if new_state == 'up' else 'disable',
-                                                                       interface, out['stderr'])
-                raise salt.exceptions.CommandExecutionError(msg)
-        return True
+        return _change_state_legacy(interface, new_state)
```

Comment:
Please move this at the top of the function, so it's easier to follow what happens for legacy systems, vs the rest. Right now is in the middle of the function.

The code should look like: 

```
if legacy:
 return _change_state_legacy()
do the non-legacy behavior
```

Reply:
Nevermind.. my bad!